### PR TITLE
De-copypastas explosive reagent reactions.

### PR DESCRIPTION
**Pull request by RandomMarine**
I wanted to add a new thing. Then I noticed these explosive reactions all had near-identical procs. So I decided this should come first.

### CODE REVIEW

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 1]
```diff
@@ -52,22 +48,18 @@
...
 	required_temp = 474
+	strength = 6
 	mix_message = "<span class='boldannounce'>Sparks start flying around the black powder!</span>"
 
-/datum/chemical_reaction/blackpowder_explosion/on_reaction(datum/reagents/holder, created_volume)
+/datum/chemical_reaction/reagent_explosion/blackpowder_explosion/on_reaction(datum/reagents/holder, created_volume)
 	sleep(rand(50,100))
-	var/location = get_turf(holder.my_atom)
-	var/datum/effect_system/reagents_explosion/e = new()
-	e.set_up(1 + round(created_volume/6, 1), location, 0, 0)
```
Comment (RandomMarine) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: And only now I notice this gets a +1. WHOOOOOOOOOPS

**Comment by Buggy123** [comment]
Sounds good, but how the heck does e.set_up work anyway? 200U of black powder, for example, results in e.set_up(200/6), or e.set_up(33), but it produces something like a 3,6,12 explosion.

The only thing I can think of is that, for some reason, it's based off of the DIAMETER of the light explosion, which is just bizarre and completely contrary to how everything else works.

**Comment by duncathan** [comment]
:+1: that's the kinda attitude I like to see

**Comment by Iamgoofball** [comment]
300u of BP is SUPPOSED to make a 3,7,14, but I think phil accidentally nerfed it with his chem redo awhile ago.

**Comment by RandomMarine** [comment]
e is the reagents_explosion effect, where set_up, as the name implies. Sets it up.
https://github.com/tgstation/-tg-station/blob/master/code/game/objects/effects/effect_system/effects_other.dm

**Comment by Buggy123** [comment]
Whoops, I misremembered the explosion testing I did.

But why 12, 6, and 3? Why set those values in the proc itself instead of balancing it by adjusting the values given to the proc?

**Pull request merged**